src/lib/Microsoft.Health.Events/EventHubProcessor/BaseEventProcessor.cs

```diff
@@ -91,6 +91,7 @@ protected virtual async Task ProcessInitializingHandler(PartitionInitializingEve
 
             try
             {
+                EventConsumerService.NewPartitionInitialized(partitionId);
```

Comment:
This will work but it would be nice to free up the partition data when the ownership is lost.  There isn't a way to detect when ownership is lost?

Reply:
yeah good point. there appears to a method `PartitionClosingAsync`  that gets called when the partition is being claimed by another processor: https://docs.microsoft.com/en-us/dotnet/api/azure.messaging.eventhubs.eventprocessorclient.partitionclosingasync?view=azure-dotnet. I can also clear the data there. 

I would like to also leave to logic at initialization to clear just in case

Reply:
Sounds good, thanks!